app/controllers/benefits/ReceiveOwnCarMileageBenefitController.scala

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.benefits
+
+import config.{AppConfig, ErrorHandler}
+import controllers.predicates.{AuthorisedAction, InYearAction}
+import forms.YesNoForm
+import models.User
+import controllers.employment.routes.CheckYourBenefitsController
+import models.employment.{BenefitsViewModel, CarVanFuelModel}
+import models.mongo.EmploymentCYAModel
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.EmploymentSessionService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import utils.{Clock, SessionHelper}
+import views.html.benefits.ReceiveOwnCarMileageBenefitView
+import services.RedirectService._
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class ReceiveOwnCarMileageBenefitController @Inject()(implicit val cc: MessagesControllerComponents,
+                                                      authAction: AuthorisedAction,
+                                                      inYearAction: InYearAction,
+                                                      receiveOwnCarMileageBenefitView: ReceiveOwnCarMileageBenefitView,
+                                                      appConfig: AppConfig,
+                                                      employmentSessionService: EmploymentSessionService,
+                                                      errorHandler: ErrorHandler,
+                                                      ec: ExecutionContext,
+                                                      clock: Clock) extends FrontendController(cc) with I18nSupport with SessionHelper {
+
+
+  def yesNoForm(implicit user: User[_]): Form[Boolean] = YesNoForm.yesNoForm(
+    missingInputError = s"benefits.receiveOwnCarMileageBenefit.error.${if (user.isAgent) "agent" else "individual"}"
+  )
+
+  def show(taxYear: Int, employmentId: String): Action[AnyContent] = authAction.async { implicit user =>
+    inYearAction.notInYear(taxYear) {
+
+      employmentSessionService.getAndHandle(taxYear, employmentId) { (optCya, _) =>
+
+        redirectBasedOnCurrentAnswers(taxYear, employmentId, optCya,
+          EmploymentBenefitsType)(commonCarVanFuelBenefitsRedirects(_, taxYear, employmentId)) { cya =>
```

Comment:
since my branch has gone in you could replace commonCarVanFuelBenefitsRedirects with mileageBenefitsRedirects